volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -279,76 +279,29 @@ def get_symbol_by_address(self, wanted_sym_address):
 
         return None
 
-    def _module_flags_taints_pre_4_10_rc1(self) -> str:
-        """Convert the module's taints value to a 1-1 character mapping.
-        Relies on statically defined taints mappings in the framework.
-
-        Returns:
-            The raw taints string.
-        """
-        taints_string = ""
-        for char, taint_flag in linux_constants.TAINT_FLAGS.items():
-            if taint_flag.module and self.taints & taint_flag.shift:
-                taints_string += char
-
-        return taints_string
-
-    def _module_flags_taints_post_4_10_rc1(self) -> str:
-        """Convert the module's taints value to a 1-1 character mapping.
-        Relies on kernel symbol embedded taints definitions.
-
-            struct taint_flag {
-                char c_true;		/* character printed when tainted */
-                char c_false;		/* character printed when not tainted */
-                bool module;		/* also show as a per-module taint flag */
-            };
-
-        Returns:
-            The raw taints string.
-        """
-        taints_string = ""
-        for i, taint_flag in enumerate(self.taint_flags_list):
-            c_true = chr(taint_flag.c_true)
-            c_false = chr(taint_flag.c_false)
-            if taint_flag.module and (self.taints & (1 << i)):
-                taints_string += c_true
-            elif taint_flag.module and c_false != " ":
-                taints_string += c_false
-
-        return taints_string
-
     def get_taints_as_plain_string(self) -> str:
         """Convert the module's taints value to a 1-1 character mapping.
+        Convenient wrapper around framework's Tainting capabilities.
 
         Returns:
             The raw taints string.
-
-        Documentation:
-            - module_flags_taint kernel function
         """
-
-        if self.taint_flags_list:
-            return self._module_flags_taints_post_4_10_rc1()
-        return self._module_flags_taints_pre_4_10_rc1()
+        return linux.Tainting(
```

Comment:
This isn't quite what I had in mind, because it requires `LinuxUtilities` to import each of the supported modules, and doesn't allow for plugins to determine whether their requested function will exist or not.  I'll have a go at mocking something up tomorrow that's more what I had in mind.  I think the separate class stuff is all good, we just need to tweak how we find the function we want to run...

Reply:
Ok, so we will have to tweak the `modules` class, to include versioned requirements of the new `Tainting` to call the tainting parsing APIs ?

https://github.com/volatilityfoundation/volatility3/blob/e7e4d76d950601258a84057779b9dd1fdff1b732/volatility3/framework/symbols/linux/extensions/__init__.py#L29

Reply:
Errr, there isn't really a way of versioning `module`, so it reverts to the version of the framework.  If it's purely an additive change (which it looks like it is) then we can just bump the MINOR version (and nothing else needs to change), but if we ever want to change it in the future, then the MAJOR version of the whole framework needs to change.  That's  why I'm so protective of things like this (and splitting them out into distinct versioned modules is good).  I think I thought these were all applying to LinuxUtilities.  I hadn't realized they were all being tacked onto `module`.  I haven't thought about how we version that separately yet.  The problem with every thing you version is that it then needs checking before use (which is painful in itself).  The requirements let us do that slightly more gracefully and finer grained, but we need to be really careful about what we add to major components of the framework and overriding the linux `module` class is one (at least at the moment)...  5:S